### DISCUSSION TITLE
Cannot get a docker image use the correct regex

### DISCUSSION THREAD

**Discussion by Tijs-2** [renovatebot/renovate#32782] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab

### Please tell us more about your question or problem

I am trying to get a version recognized for a docker image: https://hub.docker.com/r/azul/zulu-openjdk-debian
Specifically this image: azul/zulu-openjdk-debian:21.0.5-21.38-jre

So I added this:
```
packageRules: [
      {
        "matchPackageNames": ["azul/zulu-openjdk-debian"],
        "versioning": "regex:^(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)-(?<build>\\d+)\\.(?<revision>\\d+)-(?<compatibility>[a-z]+)$"
      }
]
```

There is a new version available so it should detect that, but when I run Renovate it seems to ignore my regex, and not find the new version.

No idea what I am doing wrong...
Hopefully somebody knows the answer...

### Logs (if relevant)

<details><summary>Logs</summary>

```

"config": {
         "dockerfile": [
           {
             "deps": [
               {
                 "depName": "azul/zulu-openjdk-debian",
                 "currentValue": "21.0.4-21.36-jre",
                 "replaceString": "azul/zulu-openjdk-debian:21.0.4-21.36-jre",
                 "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
                 "datasource": "docker",
                 "depType": "final",
                 "updates": [],
                 "packageName": "azul/zulu-openjdk-debian",
                 "versioning": "regex:^(?<major>\\d+)?(\\.(?<minor>\\d+))?(\\.(?<patch>\\d+))?([\\._+](?<build>(\\d\\.?)+)(LTS)?)?(-(?<compatibility>.*))?$",
                 "warnings": [],
                 "registryUrl": "https://index.docker.io/",
                 "currentVersion": "21.0.4-21.36-jre",
                 "currentVersionTimestamp": "2024-10-15T04:49:29.755Z",
                 "currentVersionAgeInDays": 44,
                 "fixedVersion": "21.0.4-21.36-jre"
               }
             ],
             "packageFile": "Dockerfile"
           }
         ],

```

</details>

**Comment by RahulGautamSingh** [comment]
You are using different versioning string in your actual repo than the one you have added in the description. Confirmed by the `versioning` in the logs.

[Repro here ](https://github.com/Rahul-renovate-testing/repro-32782) which works as expected wyhen using the correct regex.
